drivers/pmic/max170xx.c

```diff
@@ -0,0 +1,310 @@
+		&& max170xx_is_awake(drv_data, current_millis))
+	{
+		u32_t delta = current_millis - drv_data->awake_since_ms;
+		if(delta >= since_millis)
+		{
+			return;
+		} else {
+			sleep_millis -= delta;
+		}
+	} else {
+		max170xx_wake(drv_data);
+	}
+	k_sleep(sleep_millis);
+}
+
+
+
+float max170xx_soc_get(struct device *dev)
+{
+	struct max170xx_data *drv_data = dev->driver_data;
+	u16_t soc_raw;
+	max170xx_reg_read(drv_data, 0x04, &soc_raw);
+	float soc_percent = ((soc_raw >> 8) + (soc_raw & 0x00ff) / 256.f);
+	return soc_percent;
+}
+
+int max170xx_vcell_get(struct device *dev)
+{
+	struct max170xx_data *drv_data = dev->driver_data;
+
+	/* The VCELL register requires 500ms to update after exiting Sleep mode */
+	max170xx_ensure_awake(drv_data, 500);
+	u16_t volt_raw;
+	max170xx_reg_read(drv_data, 0x02, &volt_raw);
+	return (int)((volt_raw >> 4)*1.25f);
+
+}
+
+#ifndef CONFIG_MAX170XX_MAX17043
+int max170xx_chargerate_get(void)
+{
+
+	u16_t crate_raw;
+	max170xx_reg_read(&max170xx_drv_data, 0x16, &crate_raw);
+	return (int)(crate_raw * 0.208f);
+}
+#endif
+
+
+
+void max170xx_info_print(struct device *dev)
+{
+	int voltage = max170xx_vcell_get(dev);
+	LOG_DBG("%d", voltage);
+#ifdef CONFIG_MAX170XX_MAX17043
+	LOG_WRN("Reading charge rate is not supported");
+#else
+
+#endif
+	max170xx_sleep(dev);
+}
+
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int max170xx_device_ctrl(struct device *dev, u32_t ctrl_command,
+				void *context)
+{
+	struct apds9960_data *data = dev->driver_data;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		u32_t device_pm_state = *(u32_t *)context;
+
+		if (device_pm_state == DEVICE_PM_ACTIVE_STATE) {
+			if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+						APDS9960_ENABLE_REG,
+						APDS9960_ENABLE_PON,
+						APDS9960_ENABLE_PON)) {
+				return -EIO;
+			}
+
+			return 0;
+		}
+
+		if (i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+					APDS9960_ENABLE_REG,
+					APDS9960_ENABLE_PON, 0)) {
+			return -EIO;
+		}
+
+		if (i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+				       APDS9960_AICLEAR_REG, 0)) {
+			return -EIO;
+		}
+
+		return 0;
+
+	} else if (ctrl_command == DEVICE_PM_GET_POWER_STATE) {
+		*((u32_t *)context) = DEVICE_PM_ACTIVE_STATE;
+	}
+
+	return 0;
+}
+#endif
+
+int max170xx_init(struct device *dev)
+{
+	struct max170xx_data *drv_data = dev->driver_data;
+
+	drv_data->i2c = device_get_binding(drv_data->bus_name);
+
+	if (drv_data->i2c == NULL) {
+		LOG_DBG("Failed to get pointer to %s device!",
+				drv_data->bus_name);
+		return -EINVAL;
+	} else {
+		LOG_INF("MAX17043 initialized on %s", log_strdup(drv_data->bus_name));
+		max170xx_wake(drv_data);
+		k_sleep(500);
+		u16_t rev_val;
+		max170xx_reg_read(drv_data, 0x0, &rev_val);
+		LOG_DBG("MAX17043 on %s returned %u", log_strdup(drv_data->bus_name), rev_val);
+		k_sleep(1000);
+		if (rev_val == 0xc0) {
+
+		}
+	}
+
+	return 0;
+}
+
+static const struct pmic_gauge_api max170xx_gauge_api = {
```

Comment:
I think this is the part to discuss about. I don't know the best way yet to expose multiple APIs